vllm/v1/engine/output_processor.py

```diff
@@ -377,9 +387,60 @@ def propagate_error(self, e: Exception):
     def abort_requests(
         self,
         request_ids: Iterable[str],
+        internal: bool | None,
     ) -> list[str]:
-        request_ids_to_abort = []
+        """Abort a list of requests.
+
+        The request_ids may be either external request IDs (those passed to
+        InputProcessor.process_inputs()) or internal request IDs (those randomly
+        generated when creating the EngineCoreRequest).
+
+        If internal is non-None, request_ids are required to be either internal
+        or external IDs.
+
+        If an external request ID is provided, and that external request ID
+        was used for multiple requests, all requests associated with that external
+        request ID are aborted.
+
+        In the case of parallel sampling, a request ID may be used to identify
+        a parent request, in which case the associated child requests are aborted
+        also.
+        """
+
+        if internal is not None:
+            check_internal, check_external = internal, not internal
+        else:
+            check_internal, check_external = True, True
```

Comment:
Do we need to support `None` case? Thought it would be a bit simpler if we dont...

Reply:
bumping this question

Reply:
Good point, dropping that now (I had locked myself into assuming we needed it for the default, backwards compat case). Thanks